### ISSUE TITLE
Failing for 1000 decillion

### ISSUE THREAD

**Issue by NavisGames** [python-humanize/humanize#86]
### What did you do?

So, i am using Humanize in Python and Nextcord for a discord bot to shorten my Numbers with many zeros in a json file. When i am humanizing it code which is followed below

### What actually happened?, What did you expect to happen?

the user has 1000000000000000000000000000000000000 money, it says 0 googol instead of 1 googol. Adding more Zeroes will cause a failure and say then like 10000 decillion

That the user has so much money is basically impossible, but just to be safe that if the user has more than the amount of numbers,
i want to see if its changeable / fixable

### What versions are you using?

* OS:  Windows 10 
* Python: Python 3.10
* Humanize: 4.40

Please include **code** that reproduces the issue.

The [best reproductions](https://stackoverflow.com/help/minimal-reproducible-example) are [self-contained scripts](https://ericlippert.com/2014/03/05/how-to-debug-small-programs/) with minimal dependencies.

```python
import humanize

def humanizing(num):
    num = humanize.intword(num)
    num = num.replace(".0", "")
    return num



@client.command()
async def account(ctx, player: nextcord.Member = None):
    if player is None:
        player = ctx.author
    await open_account(player)
    await check(player)
    user = player
    users = await get_bank_data()
    wallet_amt = users[str(user.id)]["Wallet"]
    print(humanizing(wallet_amt))
```
```json
{
    "593456006160318464": {
        "Wallet": 1000000000000000000000000000000000000,
        "Bank": 0,
        "Bounty": 0,
        "Username": "Navis"
    }
}
```

It could be the code of mine too, but i expect it to be a issue of humanize OR it could be because humanize dont support above googol? I dont know.

**Comment by carterbox** [comment]
Could you please include *runable* code that reproduces the issue without all of the banking logic that is not relevant? Your example doesn't have a main function, so if we copy what you provided into a script it won't run.

Relevant code is here: https://github.com/python-humanize/humanize/blob/main/src/humanize/number.py#L193

**Comment by NavisGames** [comment]
Sure im sorry!
```py
import humanize


def humanizing(num):
    num = humanize.intword(num)
    num = num.replace(".0", "")
    return num


print(humanizing(1100000000000000000000000000000000000))
```
would print 0 googol

```py
import humanize


def humanizing(num):
    num = humanize.intword(num)
    num = num.replace(".0", "")
    return num


print(humanizing(1100000000000000000000000000000000000))
```
replacing the first 0 with a 1 would give 1100 decillion instead of googol.

**Comment by hugovk** [comment]
With this:
```python
import humanize

for zeroes in range(41):
    x = "1" + zeroes * "0"
    print(zeroes, "->", x, "->", humanize.intword(x))
```

For latest `main` we get:
```
0 -> 1 -> 1
1 -> 10 -> 10
2 -> 100 -> 100
3 -> 1000 -> 1.0 thousand
4 -> 10000 -> 10.0 thousand
5 -> 100000 -> 100.0 thousand
6 -> 1000000 -> 1.0 million
7 -> 10000000 -> 10.0 million
8 -> 100000000 -> 100.0 million
9 -> 1000000000 -> 1.0 billion
10 -> 10000000000 -> 10.0 billion
11 -> 100000000000 -> 100.0 billion
12 -> 1000000000000 -> 1.0 trillion
13 -> 10000000000000 -> 10.0 trillion
14 -> 100000000000000 -> 100.0 trillion
15 -> 1000000000000000 -> 1.0 quadrillion
16 -> 10000000000000000 -> 10.0 quadrillion
17 -> 100000000000000000 -> 100.0 quadrillion
18 -> 1000000000000000000 -> 1.0 quintillion
19 -> 10000000000000000000 -> 10.0 quintillion
20 -> 100000000000000000000 -> 100.0 quintillion
21 -> 1000000000000000000000 -> 1.0 sextillion
22 -> 10000000000000000000000 -> 10.0 sextillion
23 -> 100000000000000000000000 -> 100.0 sextillion
24 -> 1000000000000000000000000 -> 1.0 septillion
25 -> 10000000000000000000000000 -> 10.0 septillion
26 -> 100000000000000000000000000 -> 100.0 septillion
27 -> 1000000000000000000000000000 -> 1.0 octillion
28 -> 10000000000000000000000000000 -> 10.0 octillion
29 -> 100000000000000000000000000000 -> 100.0 octillion
30 -> 1000000000000000000000000000000 -> 1.0 nonillion
31 -> 10000000000000000000000000000000 -> 10.0 nonillion
32 -> 100000000000000000000000000000000 -> 100.0 nonillion
33 -> 1000000000000000000000000000000000 -> 1.0 decillion
34 -> 10000000000000000000000000000000000 -> 10.0 decillion
35 -> 100000000000000000000000000000000000 -> 100.0 decillion
36 -> 1000000000000000000000000000000000000 -> 0.0 googol
37 -> 10000000000000000000000000000000000000 -> 10000.0 decillion
38 -> 100000000000000000000000000000000000000 -> 100000.0 decillion
39 -> 1000000000000000000000000000000000000000 -> 1000000.0 decillion
40 -> 10000000000000000000000000000000000000000 -> 10000000.0 decillion
```

But with 1.0.0 we get:
```
0 -> 1 -> 1
1 -> 10 -> 10
2 -> 100 -> 100
3 -> 1000 -> 1000
4 -> 10000 -> 10000
5 -> 100000 -> 100000
6 -> 1000000 -> 1.0 million
7 -> 10000000 -> 10.0 million
8 -> 100000000 -> 100.0 million
9 -> 1000000000 -> 1.0 billion
10 -> 10000000000 -> 10.0 billion
11 -> 100000000000 -> 100.0 billion
12 -> 1000000000000 -> 1.0 trillion
13 -> 10000000000000 -> 10.0 trillion
14 -> 100000000000000 -> 100.0 trillion
15 -> 1000000000000000 -> 1.0 quadrillion
16 -> 10000000000000000 -> 10.0 quadrillion
17 -> 100000000000000000 -> 100.0 quadrillion
18 -> 1000000000000000000 -> 1.0 quintillion
19 -> 10000000000000000000 -> 10.0 quintillion
20 -> 100000000000000000000 -> 100.0 quintillion
21 -> 1000000000000000000000 -> 1.0 sextillion
22 -> 10000000000000000000000 -> 10.0 sextillion
23 -> 100000000000000000000000 -> 100.0 sextillion
24 -> 1000000000000000000000000 -> 1.0 septillion
25 -> 10000000000000000000000000 -> 10.0 septillion
26 -> 100000000000000000000000000 -> 100.0 septillion
27 -> 1000000000000000000000000000 -> 1.0 octillion
28 -> 10000000000000000000000000000 -> 10.0 octillion
29 -> 100000000000000000000000000000 -> 100.0 octillion
30 -> 1000000000000000000000000000000 -> 1.0 nonillion
31 -> 10000000000000000000000000000000 -> 10.0 nonillion
32 -> 100000000000000000000000000000000 -> 100.0 nonillion
33 -> 1000000000000000000000000000000000 -> 1.0 decillion
34 -> 10000000000000000000000000000000000 -> 10.0 decillion
35 -> 100000000000000000000000000000000000 -> 100.0 decillion
36 -> 1000000000000000000000000000000000000 -> 1000.0 decillion
37 -> 10000000000000000000000000000000000000 -> 10000.0 decillion
38 -> 100000000000000000000000000000000000000 -> 100000.0 decillion
39 -> 1000000000000000000000000000000000000000 -> 1000000.0 decillion
40 -> 10000000000000000000000000000000000000000 -> 10000000.0 decillion
```

Diffing the output:

![image](https://user-images.githubusercontent.com/1324225/210587934-067b62d9-21c1-418c-988d-615014fddac7.png)

Bisecting, https://github.com/jmoiron/humanize/pull/113 introduced this in 1.1.0:

```
818c9b3f1654a8f488c0c6fb828b45fdca0fade6 is the first bad commit
commit 818c9b3f1654a8f488c0c6fb828b45fdca0fade6
Author: Jasarin-V <38655357+Jasarin-V@users.noreply.github.com>
Date:   Mon Apr 23 18:22:43 2018 +0700

    fixed intword returning 1000.0 million instead of 1.0 billion

 src/humanize/number.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)
```

**Comment by hugovk** [comment]
Please see PR https://github.com/python-humanize/humanize/pull/87.
